library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -533,3 +611,44 @@ impl fmt::Display for ExitStatus {
         }
     }
 }
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+#[derive(Debug)]
+pub struct PidFd {
+    fd: libc::c_int,
+}
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+impl AsRawFd for PidFd {
+    fn as_raw_fd(&self) -> RawFd {
+        self.fd
+    }
+}
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+impl IntoRawFd for PidFd {
+    fn into_raw_fd(self) -> RawFd {
+        let fd = self.fd;
+        crate::mem::forget(self);
+        fd
+    }
+}
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+impl Drop for PidFd {
+    fn drop(&mut self) {
+        let _ = unsafe { libc::close(self.fd) };
+    }
+}
+
+#[cfg(target_os = "linux")]
+#[unstable(feature = "linux_pidfd", issue = "none")]
+impl crate::os::linux::process::ChildExt for crate::process::Child {
+    fn pidfd(&self) -> io::Result<&PidFd> {
+        self.handle.pidfd.as_ref().ok_or_else(|| Error::from(ErrorKind::Other))
```

Comment:
You can add an explanatory text here, because just `ErrorKind::Other` is not really helpful to the user:

```rust
Error::new(ErrorKind::Other, "pidfd was not requested on process creation");
```